core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -550,6 +550,44 @@ bool stm32mp_clock_is_non_secure(unsigned long clock_id)
 	return stm32mp_periph_is_non_secure(shres_id);
 }
 
+static bool mckprot_resource(enum stm32mp_shres id)
+{
+	switch (id) {
+	case STM32MP1_SHRES_MCU:
+	case STM32MP1_SHRES_PLL3:
+	case STM32MP1_SHRES_PLL3_P:
+	case STM32MP1_SHRES_PLL3_Q:
+	case STM32MP1_SHRES_PLL3_R:
+		return true;
+	default:
+		return false;
+	}
+}
+
+static void check_rcc_secure_configuration(void)
+{
+	bool secure = stm32_rcc_is_secure();
+	bool mckprot = stm32_rcc_is_mckprot();
+	enum stm32mp_shres id = STM32MP1_SHRES_COUNT;
+	unsigned int error = 0;
```

Comment:
It's a matter of taste, but I would have used a `bool have_error = false` and only assigned `true` on error.
For me the `++` operation below raises a warning flag and I need to check the code a bit closer to be able to understand what's going on.